org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/InjectConfiguration.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/*
+ * Configure the Config-Admin using the annotation `@InjectConfiguration`
+ * Get or Create an new Configuration Object.
+ * Inject the `org.osgi.service.cm.Configuration` (if used on a Parameter or Field) for further updates or delete of the `org.osgi.service.cm.Configuration`
+ *
+ */
+@Inherited
+@Target({
+	TYPE, METHOD, PARAMETER, FIELD
+})
+@Repeatable(Configurations.class)
+@Retention(RUNTIME)
+@Documented
+
+public @interface InjectConfiguration {
+
+	/**
+	 * The pid of the Configuration.
+	 *
+	 * @return The pid
+	 */
+	String pid();
+
+	/**
+	 * Indicate the name of the Factory-Configuration (not {@code null}). If
+	 * empty - auto-generated UUID will be used.
+	 *
+	 * @return The name of the Factory.
+	 */
+
+	String factoryName() default "";
+
+
+	/**
+	 * Indicate that the Configuration is a Factory-Configuration .
+	 *
+	 * @return The factory-marker.
+	 */
+
+	boolean factory() default false;
```

Comment:
The combination of these two parameters makes it possible for the user to set an inconsistent configuration - if you specify a `factoryName` but set `factory=false` (eg, you forget to set factory), what happens? One of two things should happen - either we throw an exception to alert the user to the inconsistency, or else we simply define that if `factoryName` is set to a non-empty value then `factory` is ignored.

Alternatively, it might make more sense to have `factoryName` and `autoFactory`, and then throw an exception if `factoryName` is non-empty and `autoFactory` is `true`.

Reply:
an other way would be to remove both annotations

`service.pid` - not a factory
`service.pid~` - a factory - auto generated
`service.pid~factoryName` - a factory with given factoryName

Reply:
@bjhargrave @kriegfrj could you have a look a this.